### PR TITLE
Merge into one monorepo

### PR DESCRIPTION
**Pull request by lbb**
This should replace our current api repo

TODO:
* [ ] Rename kubermatic/api -> kubermatic/kubermatic
* [x] Split secrets from config files
* [x] Merge config into kubermatic/kubermatic
* [x] Merge bare-metal-provider into kubermatic/kubermatic
* [x] Fix imports
* [x] Fix builds
* [ ] Jenkinsfile
* [ ] Test everything
* [ ] Remove symlink assumptions (i.e api/templates/coreos/...)

### CODE REVIEW

File: api/handler/cluster.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func newClusterEndpointV2(
 }
 
 // Deprecated at V2 of create cluster endpoint
-// @TODO Remove with https://github.com/kubermatic/api/issues/220
+// @TODO Remove with https://github.com/kubermatic/kubermatic/api/issues/220
```
Comment (mrIncompetent) [inline api/handler/cluster.go:103]: Should not have been modified

File: api/handler/cluster.go [hunk 2]
```diff
@@ -157,7 +157,7 @@ func clusterEndpoint(
...
```
Comment (mrIncompetent) [inline api/handler/cluster.go:160]: Should not have been modified

File: api/handler/cluster.go [hunk 3]
```diff
@@ -330,14 +330,14 @@ func createAddonEndpoint(
...
```
Comment (mrIncompetent) [inline api/handler/cluster.go:333]: Should not have been modified

File: api/handler/cluster.go [hunk 4]
```diff
@@ -330,14 +330,14 @@ func createAddonEndpoint(
...
 type newClusterReq struct {
 	dcReq
 	cluster api.Cluster
 }
 
 // Deprecated at V2 of create cluster endpoint
-// @TODO Remove with https://github.com/kubermatic/api/issues/220
+// @TODO Remove with https://github.com/kubermatic/kubermatic/api/issues/220
```
Comment (mrIncompetent) [inline api/handler/cluster.go:340]: Should not have been modified

File: api/provider/kubernetes/kubernetes.go [hunk 5]
```diff
@@ -169,7 +169,7 @@ func (p *kubernetesProvider) NewClusterWithCloud(user provider.User, spec *api.C
 }
 
 // Deprecated in favor of NewClusterWithCloud
-// @TODO Remove with https://github.com/kubermatic/api/issues/220
+// @TODO Remove with https://github.com/kubermatic/kubermatic/api/issues/220
```
Comment (mrIncompetent) [inline api/provider/kubernetes/kubernetes.go:172]: Should not have been modified

File: api/provider/kubernetes/kubernetes.go [hunk 6]
```diff
@@ -273,7 +273,7 @@ func (p *kubernetesProvider) Cluster(user provider.User, cluster string) (*api.C
...
```
Comment (mrIncompetent) [inline api/provider/kubernetes/kubernetes.go:276]: Should not have been modified

File: api/provider/kubernetes/kubernetes.go [hunk 7]
```diff
@@ -342,7 +342,7 @@ func (p *kubernetesProvider) SetCloud(user provider.User, cluster string, cloud
 // Deprecated at V2 of create cluster endpoint
 // this is a super hack and dirty hack to load the AWS cloud config from the cluster controller's templates
 // to create the config map by hand for now.
-// @TODO Remove with https://github.com/kubermatic/api/issues/220
+// @TODO Remove with https://github.com/kubermatic/kubermatic/api/issues/220
```
Comment (mrIncompetent) [inline api/provider/kubernetes/kubernetes.go:345]: Should not have been modified

File: api/test/helper.go [hunk 8]
```diff
@@ -1,16 +1,13 @@
 package test
 
-import "os"
-
 // IsOnCi returns if we're running on a ci system where no config is mounted
 func IsOnCi(path string) bool {
```
Comment (mrIncompetent) [inline api/test/helper.go:4]: Please add a issue so we wont forget to fix this (or actually remove this)

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

**Review by guusvw — APPROVED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**